homeassistant/components/stream/worker.py

```diff
@@ -60,32 +43,133 @@ def stream_worker(hass, stream, quit_event):
+                        first_packet[packet.stream].dts = first_pts[packet.stream]
+                initial_packets.append(packet)
+            if audio_stream and first_packet[audio_stream] is None:
+                _LOGGER.warning(
+                    "Audio stream not found"
+                )  # Some streams declare an audio stream and never send any packets
+                del first_pts[audio_stream]
+                audio_stream = None
+
+        except (av.AVError, StopIteration) as ex:
+            # End of stream, clear listeners and stop thread
+            for fmt, _ in outputs.items():
+                hass.loop.call_soon_threadsafe(stream.outputs[fmt].put, None)
+            _LOGGER.error(
+                "Error demuxing stream while finding first packet: %s", str(ex)
+            )
+            quit_event.set()
+
+    def initialize_segment(video_pts):
+        """Reset some variables and initialize outputs for each segment."""
+        nonlocal outputs, sequence, segment_start_pts
+        # Clear outputs and increment sequence
+        outputs = {}
+        sequence += 1
+        segment_start_pts = video_pts
+        for stream_output in stream.outputs.values():
+            if video_stream.name not in stream_output.video_codecs:
+                continue
+            buffer = create_stream_buffer(
+                stream_output, video_stream, audio_stream, sequence
+            )
+            outputs[stream_output.name] = (
+                buffer,
+                {video_stream: buffer.vstream, audio_stream: buffer.astream},
+            )
+
+    def mux_video_packet(packet):
+        # adjust pts and dts before muxing
+        packet.pts -= first_pts[video_stream]
+        packet.dts -= first_pts[video_stream]
+        # mux packets to each buffer
+        for buffer, output_streams in outputs.values():
+            # Assign the packet to the new stream & mux
+            packet.stream = output_streams[video_stream]
+            buffer.output.mux(packet)
+
+    def mux_audio_packet(packet):
+        # almost the same as muxing video but add extra check
+        # adjust pts and dts before muxing
+        packet.pts -= first_pts[audio_stream]
+        packet.dts -= first_pts[audio_stream]
+        for buffer, output_streams in outputs.values():
+            # Assign the packet to the new stream & mux
+            if output_streams.get(audio_stream):
+                packet.stream = output_streams[audio_stream]
+                buffer.output.mux(packet)
+
+    peek_first_pts()
+    last_dts = {k: v - 1 for k, v in first_pts.items()}
+    initialize_segment(first_pts[video_stream])
 
     while not quit_event.is_set():
         try:
-            packet = next(container.demux(video_stream))
+            if len(initial_packets) > 0:
+                packet = initial_packets.popleft()
+            else:
+                packet = next(container.demux((video_stream, audio_stream)))
```

Comment:
Because you are only checking for video stream below and assume anything else is audio when muxing, I also explicitly added only the video and audio streams to the demuxer here as well.